src/topology/unit_interval.lean

```diff
@@ -0,0 +1,82 @@
+/-
+Copyright (c) 2020 Patrick Massot. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Patrick Massot
+-/
+import topology.instances.real
+
+/-!
+# The unit interval, as a topological space
+
+Use `open_locale unit_interval` to turn on the notation `I := set.Icc (0 : ℝ) (1 : ℝ)`.
+
+We provide basic instances, as well as a custom tactic for discharging
+`0 ≤ x`, `0 ≤ 1 - x`, `x ≤ 1`, and `1 - x ≤ 1` when `x : I`.
+
+-/
+
+noncomputable theory
+open_locale classical topological_space filter
+open set
+
+/-! ### The unit interval -/
+
+/-- The unit interval `[0,1]` in ℝ. -/
+abbreviation unit_interval : set ℝ := set.Icc 0 1
+
+localized "notation `I` := unit_interval" in unit_interval
+
+lemma Icc_zero_one_symm {t : ℝ} : t ∈ I ↔ 1 - t ∈ I :=
+begin
+  rw [mem_Icc, mem_Icc],
+  split ; intro ; split ; linarith
+end
+
+instance I_has_zero : has_zero I := ⟨⟨0, by split ; norm_num⟩⟩
```

Comment:
Now that `unit_interval` is an abbreviation, I think it makes sense to put the following in a `namespace unit_interval`.
That also takes care of avoiding `I` in the names.

Reply:
Moved into the namespace, and renamed many declarations.